src/main/java/com/itschool/eventmanagment/services/EventServiceImp.java

```diff
@@ -40,7 +42,15 @@ public List<EventDTO> updateEvent() {
     }
 
     @Override
-    public List<EventDTO> getEvents() {
-        return null;
+    public List<EventDTO> getEvents(LocalDateTime to, LocalDateTime from, String location, String sortBy) {
```

Comment:
**Current logic:**
- this method currently in first place checks if `sortBy` is `location` -> if true, it sorts events by `location`
- otherwise, it defaults to sorting by `dateTime`

**Issus with the current logic:**
- as we have it now, we always assume that both date and location are present, and it doesn't handle cases where either `from`, `to` or `location` is null
- we don't have a default sorting - if no `sortBy` is provided, it defaults to sorting by `dateTime`, but it doesn't allow for custom behavior when no sorting is explicitly metioned
- the `OR` operator in the repository query can lead to cases where the events that match either the date range or the location are returned, which may not be the intended behavior

**Potential ideas:**
- handle missing parameters - if `from`, `to` and `location` are optional, check for their presence and adjust the query dinamically; if they are mandatory (as I assume), then we can throw a custom exception if one of them is missing
-  if `sortBy` is null or empty, default to sorting by `dateTime`
- if `soryBy` is provided but doesn't match any known values, fallback to the default sorting mechanism

**Considerations - Sorting in a Filtering API**
In RESTful design, it’s common to have sorting separated from filtering. For example:

- Filtering parameters: from, to, location (used to narrow down results based on criteria)
- Sorting parameter: sortBy (used to order the results once they are filtered)

Usually, a sortBy parameter can specify:

- The field by which to sort (e.g., date, location, name)
- The direction of sorting (asc or desc)

A more flexible approach to the sortBy parameter could look like this:
`sortBy=dateTime,desc or sortBy=location,asc`

Also, the method name can be a bit more descriptive, if you want to:
- searchEvents
- getFilteredAndSortedEvents